GomokuServer/src/GomokurServer.Core/Games/Entities/Game.cs

```diff
@@ -215,7 +215,7 @@ public virtual RematchResult Rematch(string playerId)
 			return canRematchResult;
 		}
 
-		var newGame = new Game(GameSettings, new Players(Players.White, Players.Black), _dateTimeProvider)
+		var newGame = new Game(GameSettings, Players.GetSwitchedColors(), _dateTimeProvider)
```

Comment:
I would suggest to add a null check for Players before calling GetSwitchedColors() to prevent potential NullReferenceException

Reply:
After last refactoring Game class accepts Players in constructor. Now, if Game is created, Players can't be null in this object

Reply:
Amazing, thanks for sharing insights !